openassessment/data.py

```diff
@@ -993,15 +997,26 @@ def create_zip_with_attachments(cls, file, course_id, submission_files_data):
 
         with ZipFile(file, 'w') as zip_file:
             for file_data in submission_files_data:
-                file_content = (
-                    cls._download_file_by_key(file_data['key'])
-                    if file_data['type'] == cls.ATTACHMENT
-                    else file_data['content']
-                )
+                key = file_data['key']
+                file_name = file_data['file_path']
+
+                try:
+                    file_content = (
+                        cls._download_file_by_key(file_data['key'])
+                        if file_data['type'] == cls.ATTACHMENT
+                        else file_data['content']
+                    )
+                except FileMissingException:
+                    file_data['valid'] = False
```

Comment:
I think that `file_found` (or something like that) might be a better header than `valid`, since `valid` kinda implies there's something wrong with the file. There's nothing necessarily wrong with the file itself, we just can't find it.